src/20240528_Multi_GHOCrossChainLaunch/AaveV3Arbitrum_GHOCrossChainLaunch_20240528.sol

```diff
@@ -0,0 +1,230 @@
+  }
+
+  function _deployGhoToken() internal returns (address) {
+    // Deploy imple
+    address ghoTokenImpl = ICreate3Factory(MiscArbitrum.CREATE_3_FACTORY).create(
+      GHO_IMPL_DEPLOY_SALT,
+      type(UpgradeableGhoToken).creationCode
+    );
+    require(ghoTokenImpl == GHO_IMPL, 'UNEXPECTED_GHO_TOKEN_IMPL_ADDRESS');
+
+    // proxy deploy and init
+    bytes memory ghoTokenInitParams = abi.encodeWithSignature(
+      'initialize(address)',
+      GovernanceV3Arbitrum.EXECUTOR_LVL_1 // owner
+    );
+    bytes memory creationCode = abi.encodePacked(
+      type(TransparentUpgradeableProxy).creationCode,
+      abi.encode(
+        GHO_IMPL,
+        MiscArbitrum.PROXY_ADMIN, // proxy admin
+        ghoTokenInitParams
+      )
+    );
+    return ICreate3Factory(MiscArbitrum.CREATE_3_FACTORY).create(GHO_DEPLOY_SALT, creationCode);
+  }
+
+  function _deployCcipTokenPool() internal returns (address) {
+    // Deploy imple
+    bytes memory implCreationCode = abi.encodePacked(
+      type(UpgradeableBurnMintTokenPool).creationCode,
+      abi.encode(
+        GHO, // token
+        CCIP_ARM_PROXY, // armProxy
+        false // allowlistEnabled
+      )
+    );
+    address tokenPoolImpl = ICreate3Factory(MiscArbitrum.CREATE_3_FACTORY).create(
+      CCIP_TOKEN_POOL_IMPL_DEPLOY_SALT,
+      implCreationCode
+    );
+    require(CCIP_TOKEN_POOL_IMPL == tokenPoolImpl, 'UNEXPECTED_CCIP_TOKEN_POOL_IMPL_ADDRESS');
+
+    // proxy deploy and init
+    address[] memory emptyArray = new address[](0);
+    bytes memory tokenPoolInitParams = abi.encodeWithSignature(
+      'initialize(address,address[],address)',
+      GovernanceV3Arbitrum.EXECUTOR_LVL_1, // owner
+      emptyArray, // allowList
+      CCIP_ROUTER // router
+    );
+    bytes memory creationCode = abi.encodePacked(
+      type(TransparentUpgradeableProxy).creationCode,
+      abi.encode(
+        tokenPoolImpl, // logic
+        MiscArbitrum.PROXY_ADMIN, // proxy admin
+        tokenPoolInitParams // data
+      )
+    );
+    return
+      ICreate3Factory(MiscArbitrum.CREATE_3_FACTORY).create(
+        CCIP_TOKEN_POOL_DEPLOY_SALT,
+        creationCode
+      );
+  }
+
+  function _configureCcipTokenPool(address tokenPool) internal {
+    UpgradeableTokenPool.ChainUpdate[] memory chainUpdates = new UpgradeableTokenPool.ChainUpdate[](
+      1
+    );
+    RateLimiter.Config memory rateConfig = RateLimiter.Config({
+      isEnabled: false,
+      capacity: 0,
+      rate: 0
+    });
+    chainUpdates[0] = UpgradeableTokenPool.ChainUpdate({
+      remoteChainSelector: CCIP_ETH_CHAIN_SELECTOR,
+      allowed: true,
+      outboundRateLimiterConfig: rateConfig,
+      inboundRateLimiterConfig: rateConfig
+    });
+    UpgradeableBurnMintTokenPool(tokenPool).applyChainUpdates(chainUpdates);
+  }
+
+  function _postExecute() internal override {
+    // 7. Seed Aave Pool
+    _defensiveSeed();
+  }
+
+  function _defensiveSeed() internal {
```

Comment:
An alternative for this is supply on behalf of a counterfactual contract, so once there is some aGHO burned the contract can withdraw and burn GHO. So it would be:
1. Add counterfactual as facilitator with 1 GHO as capacity
2. Call counterfactual to mint 1 GHO and supply 1 GHO to the Aave Pool
3. Decrease bucket capacity to 0
4. Counterfactual has a `unwind` function that withdraws from pool and burns GHO if there is some aGHO burned.

Pseudo-code:
```
  function withdraw() external {
    // Check address(0) is aGHO holder;
    require(
      IERC20(aGHO).balanceOf(address(0)) >= DEFENSIVE_SEED_AMOUNT,
      'NOT_ENOUGH_DEFENSIVE_SEED'
    );
    // withdraw
    uint256 amount = IERC20(aGHO).balanceOf(address(this));
    AaveV3Arbitrum.POOL.withdraw(GHO, amount, address(this));
    // burn
    IGhoToken(GHO).burn(amount);
  }
```

Happy to hear your thoughts

Reply:
Added this alternative [here](https://github.com/aave/aave-proposals-v3/pull/13)